app/components/Sidebar/components/CollectionLink.js

```diff
@@ -67,9 +77,19 @@ function CollectionLink({
   const [{ isOver, canDrop }, drop] = useDrop({
     accept: "document",
     drop: (item, monitor) => {
+      const { id, permission } = item;
       if (monitor.didDrop()) return;
       if (!collection) return;
-      documents.move(item.id, collection.id);
+      if (collection.id === item.collectionId) return;
+      if (
+        permission !== collection.permission &&
+        (permission === null || collection.permission === null)
+      ) {
```

Comment:
I think we need to make this so the modal is only triggered when the new permission is more open than the old permission, eg. only in the case of moving from `no default access` -> `read default access`.